Flaky test: `SyncedRealmTests.waitForInitialData_timeOut`

`SyncedRealmTests.waitForInitialData_timeOut`:
```
kotlin.AssertionError: Expected an exception of io.realm.kotlin.mongodb.exceptions.DownloadingRealmTimeOutException to be thrown, but was kotlin.AssertionError: Realm should not open in time
```

Seen in https://ci.realm.io/job/realm/job/realm-kotlin/job/master/464/testReport/junit/io.realm.kotlin.test.shared/SyncedRealmTests/waitForInitialData_timeOut_macos_/

This can easily be fixed by implementing the solution used in https://github.com/realm/realm-kotlin/blob/master/test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt#L203

--- a/test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt
+++ b/test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt
@@ -20,6 +20,7 @@ import io.realm.kotlin.LogConfiguration
 import io.realm.kotlin.Realm
 import io.realm.kotlin.RealmConfiguration
 import io.realm.kotlin.VersionId
+import io.realm.kotlin.entities.sync.BinaryObject
 import io.realm.kotlin.entities.sync.ChildPk
 import io.realm.kotlin.entities.sync.ParentPk
 import io.realm.kotlin.entities.sync.SyncObjectWithAllTypes
@@ -404,8 +405,11 @@ class SyncedRealmTests {
     @Test
     fun waitForInitialData_timeOut() = runBlocking {
         val partitionValue = TestHelper.randomPartitionValue()
-        val schema = setOf(ParentPk::class, ChildPk::class)
-        val objectCount = 1000 // High enough to introduce latency when download Realm initial data
+        val schema = setOf(BinaryObject::class)
+
+        // High enough to introduce latency when download Realm initial data
+        // but not too high so that we reach Atlas' transmission limit
+        val objectCount = 5
 
         // 1. Copy a valid Realm to the server
         val user1 = app.asTestApp.createUserAndLogin()
@@ -414,7 +418,7 @@ class SyncedRealmTests {
             realm.write {
                 for (index in 0 until objectCount) {
                     copyToRealm(
-                        ParentPk().apply {
+                        BinaryObject().apply {
                             _id = "$partitionValue-$index"
                         }
                     )
@@ -429,7 +433,7 @@ class SyncedRealmTests {
         val config2: SyncConfiguration = SyncConfiguration.create(user2, partitionValue, schema)
         assertNotEquals(config1.path, config2.path)
         Realm.open(config2).use { realm ->
-            val count = realm.query<ParentPk>()
+            val count = realm.query<BinaryObject>()
                 .asFlow()
                 .filter { it.list.size == objectCount }
                 .map { it.list.size }

--- a/test/sync/src/commonMain/kotlin/io/realm/kotlin/entities/sync/BinaryObject.kt
+++ b/test/sync/src/commonMain/kotlin/io/realm/kotlin/entities/sync/BinaryObject.kt
@@ -24,5 +24,5 @@ import kotlin.random.Random
 
 class BinaryObject : RealmObject {
     @PrimaryKey var _id: String = Random.nextLong().toString()
-    var binary: ByteArray = Random.nextBytes(9999999)
+    var binary: ByteArray = Random.nextBytes(999999) // don't go crazy with the size or else we risk reaching Atlas' transmission limit
 }

